src/planner/binder/query_node/plan_setop.cpp

```diff
@@ -23,14 +24,39 @@ unique_ptr<LogicalOperator> Binder::CastLogicalOperatorToTypes(vector<LogicalTyp
 	if (node->type == LogicalOperatorType::LOGICAL_PROJECTION) {
 		// "node" is a projection; we can just do the casts in there
 		D_ASSERT(node->expressions.size() == source_types.size());
+		if (node->children.size() == 1 && node->children[0]->type == LogicalOperatorType::LOGICAL_GET) {
+			// Is this a CSV Scan?
+			auto &logical_get = node->children[0]->Cast<LogicalGet>();
+			// yuck string matching, maybe there is a better way?
+			if (logical_get.function.name == "read_csv" || logical_get.function.name == "read_csv_auto") {
```

Comment:
We probably can, by adding something like an optional `table_function_type_pushdown_t` callback to the TableFunction struct, if it's set we can call it, moving most of this logic there

Does that sound reasonable?

Reply:
I would say it depends on whether other scanners face the same issues or not; I could see Parquet/Json having sth similar.

Reply:
I agree adding a callback for this is reasonable - this is useful beyond the CSV scanner (JSON and Parquet might also want to implement this).